### ISSUE TITLE
Hitting a time out error 'Error: 500 - Request timed out.' but only client side

### ISSUE THREAD

**Issue by eric-brace** [typesense/typesense-instantsearch-adapter#220]
## Description

Using typesense-instantsearch-adapter and for longer running queries getting a 500 timed out error after 5 seconds. It seems to match the timeout here int he advanced settings for algolia https://www.algolia.com/doc/api-reference/api-methods/configuring-timeouts/

For more context we're using Remix and don't have any timeouts set there. The response actually returns from the multi_search network call but the browser seems to cancel the request before it can be used by the application.

We're also using onStateChange in <InstantSearch> component. Wondering if there's a timeout setting somewhere that we've missed or if anyone has come across this before.

Typesense adapter configuration is set to 20 seconds.

```
server: {
      apiKey: ak,
      connectionTimeoutSeconds:20,
      nodes: [
        {
          host: '<host_url>',
          path:"/typesense",
          port: 443,
          protocol: 'https',
        },
      ],
    },
```

## Steps to reproduce

<img width="890" alt="Screenshot 2024-10-07 at 18 13 34" src="https://github.com/user-attachments/assets/40c62892-35de-4dd5-affa-3a093aca6afe">
<img width="687" alt="Screenshot 2024-10-07 at 18 13 59" src="https://github.com/user-attachments/assets/212f7788-a4b7-4821-8047-b9002d1e1d86">

## Expected Behavior

Expect request to our typesense server to not timeout after 5 seconds. Using curl there's no timeout.

## Actual Behavior

5 second timeout.

## Metadata

**Typesense Version**:
"typesense-instantsearch-adapter": "^2.8.0"

**OS**:

**Comment by jasonbosco** [comment]
Do you have a reverse proxy / backend that is proxying requests between the browser and Typesense? If so, I suspect that the timeout might be coming from that system. 

In the adapter itself, connectionTimeoutSeconds is the only timeout configuration.

**Comment by eric-brace** [comment]
Checked reverse proxies and don't have any set at 5 second timeout. the reason I think it might be the adapter is because I set up the same search parameters that's taking longer than 5 seconds as a regular fetch request in my app triggered by a button click and I don't get the `500 - Request timed out` error. 

Though you may be right something else is interrupting the promise chain of typesense and the adapter is throwing an error [during validate here](https://github.com/typesense/typesense-instantsearch-adapter/blob/49c65f5bd9ebe678022d8eedb1af102d108f0674/src/TypesenseInstantsearchAdapter.js#L88) but a regular fetch request is ignoring the time out potentially. But I'm pretty stumped.

One other thing to note in the network requests when I use the fetch request the Type is 'fetch' whereas the native requests made by the adapter are type 'xhr'. Not sure if that means anything.

<img width="604" alt="Screenshot 2024-10-08 at 19 12 59" src="https://github.com/user-attachments/assets/d41948a0-dfc4-4892-bac8-5b52244f2e1c">

<img width="463" alt="Screenshot 2024-10-08 at 19 13 18" src="https://github.com/user-attachments/assets/9fb2b3fd-a45f-4562-94f1-2566f0021856">

**Comment by karolzdebel** [comment]
@jasonbosco  I don't think it has to do with the reverse proxy since we can reproduce in a local environment where we're communicating directly with a local Typesense server. Some of the multi-search results return a 500 and an error message indicating a timeout. Other multi-search results part of the same request succeed. 

I cannot reproduce when embeddings aren't being used. Could it be that the timeout stems from Typesense's network requests to Open AI or something else involving embeddings?

```
{
    "code": 500,
    "error": "Request timed out."
}
```

**Comment by karolzdebel** [comment]
@jasonbosco I don't think it has to do with the reverse proxy since we can reproduce in a local environment where we're communicating directly with a local Typesense server. Some of the multi-search results return a 500 and an error message indicating a timeout. Other multi-search results part of the same request succeed.

I cannot reproduce when embeddings aren't being used. Could it be that the timeout stems from Typesense's network requests to Open AI or something else involving embeddings?

```
{
    "code": 500,
    "error": "Request timed out."
}
```

**Comment by jasonbosco** [comment]
Ah with remote embedding, there is a separate timeout on the server side when making calls out to remote embedding services. These parameters are documented here: https://typesense.org/docs/27.1/api/vector-search.html#remote-embedding-api-parameters

The default is not 5s though, it's 30s.

**Comment by karolzdebel** [comment]
@jasonbosco Setting the `remote_embedding_timeout_ms` to 30s fixed the issue based on my testing so far. Can you verify the default is actually 30s and not 5s? Might be a bug in the docs or in the code.
